core/include/detray/navigation/intersection/helix_plane_intersector.hpp

```diff
@@ -78,8 +78,17 @@ struct helix_intersector_impl<cartesian2D<algebra_t>, algebra_t> {
         const point3_type st = getter::vector<3>(sm, 0u, 3u);
 
         // Starting point on the helix for the Newton iteration
-        scalar_type s{
-            getter::norm(trf.point_to_global(mask.centroid()) - h.pos(0.f))};
+        const vector3_type dist{trf.point_to_global(mask.centroid()) -
+                                h.pos(0.f)};
+        scalar_type denom{vector::dot(sn, h.dir(0.f))};
+
+        scalar_type s{math::abs(vector::dot(sn, dist) / denom)};
```

Comment:
Should this not be calculated after the if statement? It could throw an FPE otherwise

Reply:
OK Fixed